lib/matplotlib/collections.py

```diff
@@ -289,20 +289,19 @@ def get_datalim(self, transData):
                     self.get_transforms(),
                     offset_trf.transform_non_affine(offsets),
                     offset_trf.get_affine().frozen())
-            if has_offsets:
-                # this is for collections that have their paths (shapes)
-                # in physical, axes-relative, or figure-relative units
-                # (i.e. like scatter). We can't uniquely set limits based on
-                # those shapes, so we just set the limits based on their
-                # location.
-
-                offsets = (offset_trf - transData).transform(offsets)
-                # note A-B means A B^{-1}
-                offsets = np.ma.masked_invalid(offsets)
-                if not offsets.mask.all():
-                    bbox = transforms.Bbox.null()
-                    bbox.update_from_data_xy(offsets)
-                    return bbox
+
```

Comment:
Should the check for has_offsets a few lines earlier also be dropped? (if offset_trf is "anything", it certainly can map (0, 0) to anywhere and therefore autoscaling based on the collection probably doesn't make sense either.

Regardless of this comment, the PR as is fixes the original issue; this is just about a more general case.

Reply:
:+1: I got rid of that entire block above.